.github/workflows/gradle.yml

```diff
@@ -25,49 +25,60 @@ jobs:
   build:
     name: Build for JDK ${{ matrix.java }}
     runs-on: ubuntu-latest
-    if: ${{ github.event_name == 'push' || github.event.pull_request.head.repo.full_name != 'Discord4J/Discord4J' }}
+    if: ${{ github.event_name == 'push' || (github.event.pull_request.head.repo.full_name != 'Discord4J/Discord4J' && !contains(toJson(github.event.pull_request.labels.*.name), 'PR depends on PR')) }}
     strategy:
       matrix:
         java: [ 8, 11 ]
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/cache@v1
+      - uses: actions/checkout@v3
+      - uses: actions/cache@v3
         with:
           path: ~/.gradle/caches
           key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
           restore-keys: |
             ${{ runner.os }}-gradle-
       - name: Set up JDK ${{ matrix.java }}
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v3
         with:
           java-version: ${{ matrix.java }}
+          distribution: 'corretto'
       - name: Build with Gradle
         run: ./gradlew build
-        if: |
-          !contains(github.event.head_commit.message, '[ci skip]') && !contains(github.event.pull_request.title, '[ci skip]')
   release:
     name: Publish artifacts
     runs-on: ubuntu-latest
-    if: github.event_name != 'pull_request'
+    if: ${{ github.event_name != 'pull_request' && github.repository != 'Discord4J/Discord4J' }}
     needs: build
+    env:
+      HAVE_SONAR_DATA: ${{ secrets.signingKey != '' && secrets.signingPassword != '' && secrets.sonatypeUsername != '' && secrets.sonatypePassword != '' }}
```

Comment:
Not sure I agree with this one, should it be enough to have the `if` check?

Reply:
i need test (test in forks its a little hard) because when make this if not add the if in every step the flow still running and get error... but can test if the first not work not handle the rest,,,

Reply:
Ok i remember why cannot check this in a IF directly... based in GitHub docs you cannot use secrets in if unless you pass in env for this i make this when start this PR.
https://docs.github.com/en/actions/security-guides/encrypted-secrets#using-encrypted-secrets-in-a-workflow

Reply:
Tested and works.
not a big fan for let this in failed mostly because the emails... but works.
![image](https://user-images.githubusercontent.com/3602279/230732901-38184fc9-c09f-4e43-a9be-78f884874d43.png)

Reply:
Looking again at this, why disallow publishing to a sonatype repository even if it's a fork? Like quanticc/Discord4J or Doc94/Discord4J?

Maybe we should have something to skip the job entirely instead of failing it or disabling it for forks. Like an env var that's explicitly set if you don't want releases

Reply:
hmmm i can keep the errors for the missing vars but add a var using the Repository for allow disable the releases on push or allow a override using [release] in commit like [ci skip].

Reply:
Added, now the workflow check a var (default to true) for allow release in any commit or if the commit has the "[release]" in commit

Reply:
Thanks, we just need something minor about the wording of the message since it's `¡Release Failed!` it should not include the `¡!` exclamation points.

And the error content could improve wording: `This repository does not have all required secrets: `

Finally, use "HAS_CREDENTIALS" as env var name. Sonar is unrelated to Sonatype, the name doesn't apply here

Reply:
Ready this.